### DISCUSSION TITLE
Change Durable Function connection/taskhub name during runtime

### DISCUSSION THREAD

**Discussion by hctan** [Azure/azure-functions-durable-extension#2058] [Q&A]
Hi,

Can I change the connection or taskhub name before I start a new orchestration using the durable client? I have an use case where I need to have the durable function running in different queues based on the load at that time. 

For example, I have a function with httptrigger, it receives a request, and based on some criteria, I would like to start the orchestration on different queue. It's quite similar to what RaiseEventAsync() command can do, but on StartNewAsync().

**Comment by hctan** [comment]
My colleague found a way where we could instantiate the IDurableOrchestrationClient during runtime using the IDurableClientFactory.createClient() from the library. I guess its ok to use this to achieve what I need?

**Reply by cgillum** [reply]
That sounds good to me. Can you share a code sample showing how you're doing this?

**Comment by hctan** [comment]
Hi, here's the sample code from my colleague:

```
    public class HttpStart
    {
        private readonly IEnumerable<IDurableClient> clients;
        private readonly ILogger logger;

        public HttpStart(ILoggerFactory loggerFactory, IDurableClientFactory clientFactory)
        {
            clients = Enumerable.Range(1, 5)
                .Select(@int => clientFactory.CreateClient(new DurableClientOptions
                {
                    ConnectionName = $"DurableFunctions{@int}Storage",
                    TaskHub = "AdfPlayground",
                    IsExternalClient = true,
                }));
            logger = loggerFactory.CreateLogger<HttpStart>();
        }

        [FunctionName(nameof(HttpStart))]
        public async Task<IActionResult> RunAsync(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "instances")] HttpRequestMessage req)
        {
            NameValueCollection queryString = req.RequestUri.ParseQueryString();
            string instanceId = queryString?.GetValues("instanceId")?.FirstOrDefault() ?? Guid.NewGuid().ToString();

            await clients.ElementAt(0).StartNewAsync("Orchestrator", instanceId);

            logger.LogInformation($"Started orchestration with ID = '{instanceId}'.");

            return new OkObjectResult(new
            {
                instanceId,
            });
        }
    }
```

The client enumeration is just a sample. We'll keep all the connection name string in the azure function's configuration.
